### DISCUSSION TITLE
Is it possible to not encode ALL prompts before start?

### DISCUSSION THREAD

**Discussion by veneamin** [lllyasviel/Fooocus#2729] [Q&A] [answered]
Is it possible to not encode ALL prompts before start?
I set max image count to 1000 in config and have to wait until fooocus encoded all 1000 negative and positive prompts. Moreover, this stage is unstoppable and if change my mind during wait - i still have to wait until the end.
Is it possible to encode this stuff  right before processing image? 
And the option "generate forever" can be done in this case.

**Comment by mashb1t** [comment, marked as answer]
Feel free to adjust the code to your liking, but i'd suggest to use generate forever with a lower image count instead, e.g. 10.

**Reply by veneamin** [reply]
My programming level is not sufficient to do this myself.
Is “generate forever” already somehow available in the current version?

**Reply by mashb1t** [reply]
It is, you can select generate forever by right-clicking the generate button.

**Reply by veneamin** [reply]
Wow! its great! Thanks!
Just a little note: then i have selected wrong model (non sdxl) - i got error message in console, not in web interface. And if  I set "generate forever" with wrong model - i got disco party of buttons,  it was very hard to manage stop cycling

**Reply by DavidDragonsage** [reply]
I appreciate the efficiency of doing the encoding at the start, but Stop should do just that.

I have a slow PC (low VRAM) and so I will set it for say 30 generations - because I'm going away to do something else.

But then if I notice a mistake in the prompt, its quite painful waiting for all that encoding without any way to stop it - short of terminating Fooocus completely. But if I terminate Fooocus then I have to go through all the setup again. It can be quite frustrating.

Generate Forever is not really practical for me - I don't really want to come back to potentially sift through a hundred-plus images either.

No means no, and Stop should mean stop.
